frame/proxy/src/lib.rs

```diff
@@ -137,84 +89,84 @@ pub struct Announcement<AccountId, Hash, BlockNumber> {
-		ProxyType = <T as Config>::ProxyType,
-		Hash = CallHashOf<T>,
-	{
-		/// A proxy was executed correctly, with the given \[result\].
-		ProxyExecuted(DispatchResult),
-		/// Anonymous account has been created by new proxy with given
-		/// disambiguation index and proxy type. \[anonymous, who, proxy_type, disambiguation_index\]
-		AnonymousCreated(AccountId, AccountId, ProxyType, u16),
-		/// An announcement was placed to make a call in the future. \[real, proxy, call_hash\]
-		Announced(AccountId, AccountId, Hash),
-	}
-}
+		/// The overarching call type.
+		type Call: Parameter + Dispatchable<Origin=Self::Origin, PostInfo=PostDispatchInfo>
+			+ GetDispatchInfo + From<frame_system::Call<Self>> + IsSubType<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::Call>;
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		type Error = Error<T>;
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
 
-		/// Deposit one of this module's events by using the default implementation.
-		fn deposit_event() = default;
+		/// A kind of proxy; specified with the proxy and passed in to the `IsProxyable` fitler.
+		/// The instance filter determines whether a given call may be proxied under this type.
+		///
+		/// IMPORTANT: `Default` must be provided and MUST BE the the *most permissive* value.
+		type ProxyType: Parameter + Member + Ord + PartialOrd + InstanceFilter<<Self as Config>::Call>
+			+ Default;
 
 		/// The base amount of currency needed to reserve for creating a proxy.
-		const ProxyDepositBase: BalanceOf<T> = T::ProxyDepositBase::get();
+		///
+		/// This is held for an additional storage item whose value size is
+		/// `sizeof(Balance)` bytes and whose key size is `sizeof(AccountId)` bytes.
+		#[pallet::constant]
+		type ProxyDepositBase: Get<BalanceOf<Self>>;
 
 		/// The amount of currency needed per proxy added.
-		const ProxyDepositFactor: BalanceOf<T> = T::ProxyDepositFactor::get();
+		///
+		/// This is held for adding 32 bytes plus an instance of `ProxyType` more into a pre-existing
+		/// storage value.
+		#[pallet::constant]
+		type ProxyDepositFactor: Get<BalanceOf<Self>>;
 
 		/// The maximum amount of proxies allowed for a single account.
-		const MaxProxies: u16 = T::MaxProxies::get();
+		#[pallet::constant]
+		type MaxProxies: Get<u16>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
 
-		/// `MaxPending` metadata shadow.
-		const MaxPending: u32 = T::MaxPending::get();
+		/// The maximum amount of time-delayed announcements that are allowed to be pending.
+		#[pallet::constant]
+		type MaxPending: Get<u32>;
 
-		/// `AnnouncementDepositBase` metadata shadow.
-		const AnnouncementDepositBase: BalanceOf<T> = T::AnnouncementDepositBase::get();
+		/// The type of hash used for hashing the call.
+		type CallHasher: Hash;
+
+		/// The base amount of currency needed to reserve for creating an announcement.
```

Comment:
```suggestion
		/// The base amount of currency needed in reserve for creating an announcement.
```

Reply:
Correct me if you think I am wrong but I think the current version makes more sense. When you make an announcement you will need "to" reserve at least this amount (so it would move from free => reserve balance). But the balance is not already "in" reserve prior to making the announcement.

Reply:
yes i agree